### PR TITLE
skytemple: 1.3.10 -> 1.4.7

### PR DESCRIPTION
**Pull request by marius851000**
###### Description of changes

I updated skytemple and it’s dependancies, as well as adding newly required dependancies. I also fixed the gql package, that previously failed to compile (gradient, it’s only dependancy before skytemple doesn’t seems to compile. Despite the checks failing, the feature that depend on it in SkyTemple work with no problem).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

I’ll mention the co-maintainer @xfix as well as the maintainers of gql @fabaff

superseed https://github.com/NixOS/nixpkgs/pull/193000

### CODE REVIEW

**Comment by marius851000** [comment]
Can someone help me figure what “duplicate formal function argument 'backoff'“ means?

File: pkgs/development/python-modules/gql/default.nix [hunk 1]
```diff
@@ -41,9 +42,10 @@ buildPythonPackage rec {
     urllib3
     websockets
     yarl
+    backoff
```
Comment (fabaff) [inline pkgs/development/python-modules/gql/default.nix:45]: `backoff` was added with https://github.com/NixOS/nixpkgs/commit/e944d4fe3dea48360f3efa2ac9301438da0a3927.

File: pkgs/development/python-modules/gql/default.nix [hunk 2]
```diff
@@ -62,7 +64,11 @@ buildPythonPackage rec {
     # Exclude linter tests
     "gql-checker/tests/test_flake8_linter.py"
     "gql-checker/tests/test_pylama_linter.py"
-  ];
+  ];*/
+
+  #For some reason, seemingly related to async, or maybe the network interface, checks does not work
+
+  doCheck = false;
```
Comment (fabaff) [inline pkgs/development/python-modules/gql/default.nix:71]: The tests are passing on `master`. I guess that you branch is off as https://github.com/NixOS/nixpkgs/commit/e944d4fe3dea48360f3efa2ac9301438da0a3927 introduced adjustments to the check section.

File: pkgs/development/python-modules/pmdsky-debug-py/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    rev = version;
+    sha256 = "sha256-kbubOolFsPZko/LULPxoHCfofdefo6OtfuYO0xpjOCw=";
+  };
+
+  prePatch = "cd src";
+
+  format = "pyproject";
+
+  meta = with lib; {
+    description = "Autogenerated and statically check-able pmdsky-debug symbol definitions for Python.";
```
Comment (fabaff) [inline pkgs/development/python-modules/pmdsky-debug-py/default.nix:22]:
```suggestion
    description = "Autogenerated and statically check-able pmdsky-debug symbol definitions for Python";
```

See https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes

File: pkgs/development/python-modules/pmdsky-debug-py/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    homepage = "https://github.com/SkyTemple/pmdsky-debug-py";
+    license = [ licenses.mit ];
```
Comment (fabaff) [inline pkgs/development/python-modules/pmdsky-debug-py/default.nix:24]:
```suggestion
    license = licenses.mit;
```

File: pkgs/development/python-modules/range-typed-integers/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{ lib, buildPythonPackage, fetchFromGitHub, pytestCheckHook }:
```
Comment (fabaff) [inline pkgs/development/python-modules/range-typed-integers/default.nix:1]: Easier to review and better readable diffs in the future.

```suggestion
{ lib
, buildPythonPackage
, fetchFromGitHub
, pytestCheckHook
}:
```

File: pkgs/development/python-modules/range-typed-integers/default.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+buildPythonPackage rec {
+  pname = "range-typed-integers";
+  version = "1.0.1";
```
Comment (fabaff) [inline pkgs/development/python-modules/range-typed-integers/default.nix:5]: Only Python > 3.8 is [supported](https://github.com/theCapypara/range-typed-integers/blob/1.0.1/pyproject.toml#L6).

```suggestion
  version = "1.0.1";
  format = "pyproject";

  disabled = pythonOlder "3.8";
```

File: pkgs/development/python-modules/range-typed-integers/default.nix [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchFromGitHub {
+    owner = "theCapypara";
+    repo = "range-typed-integers";
+    rev = version;
+    sha256 = "sha256-4+XdalHq6Q2cBbuYi4x7kmCNQh1MwYf+XlLP9FzzzgE=";
+  };
+
+  checkInputs = [ pytestCheckHook ];
+
+  format = "pyproject";
```
Comment (fabaff) [inline pkgs/development/python-modules/range-typed-integers/default.nix:16]: Move this up please.

File: pkgs/development/python-modules/range-typed-integers/default.nix [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+
+  meta = with lib; {
+    description = "A package provides integer types that have a specific range of valid values";
+    homepage = "https://github.com/theCapypara/range-typed-integers";
+    license = [ licenses.mit ];
```
Comment (fabaff) [inline pkgs/development/python-modules/range-typed-integers/default.nix:21]:
```suggestion
    license = licenses.mit;
```

File: pkgs/development/python-modules/skytemple-files/default.nix [hunk 9]
```diff
@@ -21,7 +24,12 @@ buildPythonPackage rec {
 
   buildInputs = [ armips ];
 
-  propagatedBuildInputs = [ appdirs dungeon-eos explorerscript ndspy pillow setuptools skytemple-rust tilequant ];
+  propagatedBuildInputs = [
+    appdirs dungeon-eos explorerscript ndspy pillow setuptools skytemple-rust
+    tilequant pyyaml pmdsky-debug-py
+    # SpriteCollab’s specific dependancies
+    aiohttp lru-dict graphql-core gql
+  ];
```
Comment (fabaff) [inline pkgs/development/python-modules/skytemple-files/default.nix:32]: `optional-dependencies` help to separate the dependencies. Also, `typing-extensions` is [needed](https://github.com/SkyTemple/skytemple-files/blob/1.4.1/setup.py#L48).

```suggestion
  propagatedBuildInputs = [
    appdirs
    dungeon-eos
    explorerscript
    ndspy
    pillow
    pmdsky-debug-py
    pyyaml
    setuptools
    skytemple-rust
    tilequant
  ] ++ lib.optionals (pythonOlder "3.9") [
    typing-extensions
  ];

  passthru.optional-dependencies = {
    spritecollab = [
      aiohttp
      gql
      graphql-core
      lru-dict
    ] ++ gql.optional-dependencies.aiohttp;
  };
```

Depends on https://github.com/NixOS/nixpkgs/pull/200163

File: pkgs/development/python-modules/range-typed-integers/default.nix (hunk 5 above)
Comment (marius851000) [reply pkgs/development/python-modules/range-typed-integers/default.nix:1]: Do you recommend me doing this for all of the other packages I added?

Comment (marius851000) [reply pkgs/development/python-modules/range-typed-integers/default.nix]: Also, do you know if keeping , in the line of the expression is important? I have some difficulty reading , at the beginning of a line.

Comment (marius851000) [reply pkgs/development/python-modules/range-typed-integers/default.nix]: *Well, if that’s what nixpkgs-fmt use, I’ll use it. I actually look better without the red background for change*

**Comment by marius851000** [comment]
Thanks for all of those suggestion. Should be ready for review again.

**Comment by marius851000** [comment]
I’ll update this package to incldue 1.4.2 as well as enabling checks in skytemple-files, which can be run without the ROM now, in a few hours.

**Comment by xfix** [comment]
Please solve the merge conflicts (sorry it took this long for me to look into this pull request).

**Comment by marius851000** [comment]
Ping @xfix

**Review by sugar700 — APPROVED** [review]
Did some basic testing on Fedora 38 (not NixOS), checked HTML pages within editor, and used embedded emulator. Everything works fine.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 200020` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>38 packages built:</summary>
  <ul>
    <li>python310Packages.explorerscript</li>
    <li>python310Packages.explorerscript.dist</li>
    <li>python310Packages.pmdsky-debug-py</li>
    <li>python310Packages.pmdsky-debug-py.dist</li>
    <li>python310Packages.py-desmume</li>
    <li>python310Packages.py-desmume.dist</li>
    <li>python310Packages.range-typed-integers</li>
    <li>python310Packages.range-typed-integers.dist</li>
    <li>python310Packages.skytemple-dtef</li>
    <li>python310Packages.skytemple-dtef.dist</li>
    <li>python310Packages.skytemple-files</li>
    <li>python310Packages.skytemple-files.dist</li>
    <li>python310Packages.skytemple-rust</li>
    <li>python310Packages.skytemple-rust.dist</li>
    <li>python310Packages.skytemple-ssb-debugger</li>
    <li>python310Packages.skytemple-ssb-debugger.dist</li>
    <li>python310Packages.tilequant</li>
    <li>python310Packages.tilequant.dist</li>
    <li>python311Packages.explorerscript</li>
    <li>python311Packages.explorerscript.dist</li>
    <li>python311Packages.pmdsky-debug-py</li>
    <li>python311Packages.pmdsky-debug-py.dist</li>
    <li>python311Packages.py-desmume</li>
    <li>python311Packages.py-desmume.dist</li>
    <li>python311Packages.range-typed-integers</li>
    <li>python311Packages.range-typed-integers.dist</li>
    <li>python311Packages.skytemple-dtef</li>
    <li>python311Packages.skytemple-dtef.dist</li>
    <li>python311Packages.skytemple-files</li>
    <li>python311Packages.skytemple-files.dist</li>
    <li>python311Packages.skytemple-rust</li>
    <li>python311Packages.skytemple-rust.dist</li>
    <li>python311Packages.skytemple-ssb-debugger</li>
    <li>python311Packages.skytemple-ssb-debugger.dist</li>
    <li>python311Packages.tilequant</li>
    <li>python311Packages.tilequant.dist</li>
    <li>skytemple</li>
    <li>skytemple.dist</li>
  </ul>
</details>

**Review by fabaff — APPROVED** [review]
Change looks good to me.

**Pull request merged**